just/bluefin-system.just

```diff
@@ -0,0 +1,155 @@
+# Run a one minute system benchmark
+benchmark:
+    echo 'Running a 1 minute benchmark ...'
+    stress-ng --matrix 0 -t 1m --times
+
+# Configure Bluefin-CLI Terminal Experience
+bluefin-cli:
+    @/usr/libexec/enable-bluefin-cli.sh
+
+# Configure Terminal Experience
+[private]
+configure-terminal:
+    @/usr/libexec/configure-terminal.sh
+
+# Toggle between Bluefin and the Developer Experience
+devmode:
+    #!/usr/bin/env bash
+    CURRENT_IMAGE=$(rpm-ostree status -b --json | jq -r '.deployments[0]."container-image-reference"')
+    if grep -q "/var/ublue-os/image" <<< $CURRENT_IMAGE
+    then
+        echo ""
+        echo "Before we can switch to the Bluefin Developer Experience"
+        echo "the current system needs an update. Please run 'just update'"
+        echo "and reboot your system when the update is finished."
+        exit
+    fi
+    if grep -q "bluefin-dx" <<< $CURRENT_IMAGE
+    then
+        CURRENT_STATE="enabled"
+    else
+        CURRENT_STATE="disabled"
+    fi
+    echo "Developer mode is currently ${CURRENT_STATE}"
+    echo "Enable or Disable developer mode"
+    OPTION=$(gum choose Enable Disable)
+    if [ "$OPTION" = "Enable" ]
+    then
+        if [ "$CURRENT_STATE" = "enabled" ]
+        then
+            echo "You are already on a developer image"
+        else
+            echo "Rebasing to a developer image"
+            NEW_IMAGE=$(echo $CURRENT_IMAGE | sed "s/bluefin/bluefin-dx/")
+            rpm-ostree rebase $NEW_IMAGE
+        fi
+    elif [ "$OPTION" = "Disable" ]
+    then
+        if [ "$CURRENT_STATE" = "enabled" ]
+        then
+            echo "Rebasing to a non developer image"
+            NEW_IMAGE=$(echo $CURRENT_IMAGE | sed "s/bluefin-dx/bluefin/")
+            rpm-ostree rebase $NEW_IMAGE
+        else
+            echo "You are currently not on a developer image"
+        fi
+    fi
+
+alias shell := configure-shell
+
+# Switch to a different shell
+configure-shell ACTION="":
+    #!/bin/bash
+    source /usr/lib/ujust/ujust.sh
+    CURRENT_SHELL="$(cat /etc/passwd | grep ":$UID:" | cut '-d:' '-f7')"
+    OPTION={{ ACTION }}
+    if [ "$OPTION" == "help" ]; then
+      echo "Usage: ujust shell <option>"
+      echo "  <option>: Specify the quick option to skip the prompt"
+      echo "  Use 'fish' to select fish"
+      echo "  Use 'zsh' to select zsh"
+      echo "  Use 'bash' to select bash"
+      exit 0
+    elif [ "$OPTION" == "" ]; then
+      echo "${bold}Shell configuration${normal}"
+      echo "${USER}'s shell is currently ${CURRENT_SHELL}"
+      OPTION=$(Choose "fish" "zsh" "bash")
+    fi
+    if [[ "$OPTION" =~ zsh|fish ]]; then
+      sudo usermod $USER --shell /usr/bin/$OPTION 
+      printf "${USER}'s shell is now %s.\n" "$(cat /etc/passwd | grep ":$UID:" | cut '-d:' '-f7')"
+    elif [ "$OPTION" == "bash" ]; then
+      sudo usermod $USER --shell /bin/$OPTION 
+      printf "${USER}'s shell is now %s.\n" "$(cat /etc/passwd | grep ":$UID:" | cut '-d:' '-f7')"
+    fi
```

Comment:
Why are we doing an if - elif here? Also how is this handling a cancel out during the shell selection? Option becomes empty string then?

Reply:
If the string is empty it will just exit as it does not match, as there is no code coming after it.
Also i saw by default the shell is `/bin/bash` and not `/usr/bin/bash`
I know they are the same path due to symlinks but i did it more for my OCD and can just put them all together to `/usr/bin/$OPTION` :sweat_smile: